src/features/commands/character/character-command.service.ts

```diff
@@ -15,24 +15,31 @@ export class CharacterCommandService {
     return CharacterCommandService._instance;
   }
 
-  public message(message: Message): Promise<void> {
+  public message(message: Message): Promise<Message> {
     return CharacterService.getInstance()
       .getEntity(message.author.id)
-      .then((characterFound): void => {
-        const boldCharacterName = MessageFormattingService.getInstance().format(
-          TextFormatsEnum.BOLD,
-          characterFound.name
-        );
-        void DisplayMessageService.getInstance().message(
-          message,
-          `Your character's name is: ${boldCharacterName}`
-        );
-      })
-      .catch((): void => {
-        void DisplayMessageService.getInstance().message(
-          message,
-          `Character not found.`
-        );
-      });
+      .then(
+        (characterFound): Promise<Message> => {
+          if (characterFound) {
+            const boldCharacterName = MessageFormattingService.getInstance().format(
+              TextFormatsEnum.BOLD,
+              characterFound.name
+            );
+            return Promise.resolve(

```

Comment:
This is not the way to handle promises and if it works then I do not know this syntaxe and I do not like it because there is duplicated code.

Reply:
I think that you do not have a choice here, you HAVE to reject your own error when the character was not found.

Reply:
I just noticed I handled it in two different ways. But... What did you have in mind, then? °°

Reply:
```typescript
if (...) return MessageFormattingService.getInstance().format(...);
else return Promise.reject(new Error('Could not find the character from the given message));
```